### PR TITLE
Ports/psoc6 setup scripts

### PR DESCRIPTION
**Pull request by jaenrig-ifx**
This pull request provide the following:
* Batch Window and  Bash Linux script for quick start, device setup and firmware deployment.
* Refactoring of build-deploy-test-release ci workflow reusing the script to avoid rebuilding on hil improving performance. 

Please, check if these commands work for you in Windows and Linux
Windows:
```
curl.exe -s -L https://raw.githubusercontent.com/jaenrig-ifx/micropython/ports/psoc6_setup-scripts/tools/psoc6/mpy-psoc6.cmd > mpy-psoc6.cmd
mpy-psoc6.cmd quick-start

```

Linux:
```
curl -s -L https://raw.githubusercontent.com/jaenrig-ifx/micropython/ports/psoc6_setup-scripts/tools/psoc6/mpy-psoc6.sh > mpy-psoc6.sh 
chmod +x mpy-psoc6.sh 
./mpy-psoc6.sh quick-start
```

If you are wsl or such without graphic desktop support, the arduino-lab launch should fail. But you can check with minicom
```
minicom -D /dev/ttyACM0
``` 

I will add the docs in a separate pull request, with this commands pointing out to the main branch.

### CODE REVIEW

File: .github/workflows/ports_psoc6.yml [hunk 1]
```diff
@@ -11,60 +11,72 @@ on:
       - 'lib/**'
       - 'ports/psoc6/**'
 
+
```
Comment (bargfred) [inline .github/workflows/ports_psoc6.yml:14]: Just magic !

**Review by bargfred — APPROVED** [review]
Good job !

**Comment by NikhitaR-IFX** [comment]
> This pull request provide the following:
> 
> * Batch Window and  Bash Linux script for quick start, device setup and firmware deployment.
> * Refactoring of build-deploy-test-release ci workflow reusing the script to avoid rebuilding on hil improving performance.
> 
> Please, check if these commands work for you in Windows and Linux Windows:
> 
> ```
> curl.exe -s -L https://raw.githubusercontent.com/jaenrig-ifx/micropython/ports/psoc6_setup-scripts/tools/psoc6/mpy-psoc6.cmd > mpy-psoc6.cmd
> mpy-psoc6.cmd quick-start
> ```
> 
> Linux:
> 
> ```
> curl -s -L https://raw.githubusercontent.com/jaenrig-ifx/micropython/ports/psoc6_setup-scripts/tools/psoc6/mpy-psoc6.sh > mpy-psoc6.sh 
> chmod +x mpy-psoc6.sh 
> ./mpy-psoc6.sh quick-start
> ```
> 
> If you are wsl or such without graphic desktop support, the arduino-lab launch should fail. But you can check with minicom
> 
> ```
> minicom -D /dev/ttyACM0
> ```
> 
> I will add the docs in a separate pull request, with this commands pointing out to the main branch.

For windows, I am getting the below message and arduino lab does not open. Expected?
windows

In WSL, it works like a charm! 🎉🤩

**Review by IFX-Anusha — COMMENTED** [review]
In windows Arduino Lab is opening for me 🎉. But in between the following error is there
openocd

In wsl It works!!🎉

**Review by NikhitaR-IFX — APPROVED** [review]

**Review by IFX-Anusha — APPROVED** [review]

**Review by boramonideep — APPROVED** [review]

**Pull request merged**